src/msg/async/ProtocolV2.cc

```diff
@@ -1580,6 +1584,16 @@ CtPtr ProtocolV2::throttle_dispatch_queue() {
           << connection->dispatch_queue->dispatch_throttler.get_current() << "/"
           << connection->dispatch_queue->dispatch_throttler.get_max()
           << " failed, just wait." << dendl;
+      ceph::mono_time throttle_now = ceph::mono_clock::now();
+      auto duration = std::chrono::duration_cast<std::chrono::seconds>(throttle_now - throttle_prev);
+      if (duration >= msgr->dispatch_throttle_log_interval) {
+        ldout(cct, 1) << __func__ << " Throttler Limit has been hit. "
+                      << "Some message processing may be significantly delayed." << dendl;
+        throttle_prev = throttle_now;
+
+        //Cluster logging that throttling is occurring.
+        msgr->ms_deliver_throttle(ms_throttle_t::DISPATCH_QUEUE);
```

Comment:
I would think this line would be called outside of this `if` block? Wouldn't we deliver the throttle notification to the `msgr` every time?

Reply:
No, this is to limit the number of logs by configuring `dispatch_throttle_log_interval`. So if `ms_dispatch_throttle_log_interval` is say 30 seconds, the `msgr` would get notified every 30 seconds and logs the cluster log. I think we may have a better log text for the cluster log.

Reply:
No. I think you do want to deliver the throttle to the messenger dispatchers every time and leave it up to them how to respond. You have two messages to throttle using two configs: (a) a debug warning at high verbosity (0 probably) every 30 seconds and (b) a cluster warning.

Reply:
I've fixed it. The number of cluster warnings has increased largely with the fix, 57145 this time.